scripts/src/javatests/org/oppia/android/scripts/common/BazelClientTest.kt

```diff
@@ -388,7 +388,7 @@ class BazelClientTest {
       """
       package com.example
       
-      class TwoSum {
+      class AddNums {
```

Comment:
Are there existing tests for: 
Missing coverage data file.
Non-zero exit code from the Bazel command execution.?

Reply:
Hmm... We do have a case for a **non-zero exit code** in [**CoverageRunnerTest**](https://github.com/oppia/oppia-android/pull/5459/files?diff=unified&w=0#diff-2afef309affdf8a1c7137b6d8dc5a6f759f3fc35faac96ba473fbb0a89397d6cR63) as it directly interacts with the Bazel execution command and returns an exit code for an invalid test target. 

However, in [**RunCoverageTest**](https://github.com/oppia/oppia-android/pull/5459/files?diff=unified&w=0#diff-5a779ea4a8aa2caf5249ced27465f76b8ac19bed5cf250475f6033dfad464908R85), any invalid file inputs are immediately caught with an **IllegalStateException** stating **"File doesn't exist."** While RunCoverageTest does not cover the non-zero exit code, it does include the `testRunCoverage_invalidFile_throwsException` case. 

Please let me know if there are any additional considerations or alternative methods for handling this.

Reply:
@adhiamboperes, added test cases for Missing Coverage that might occur in 2 scenarios:
1. [testRunWithCoverageAsync_coverageRetrievalFailed_throwsException](https://github.com/oppia/oppia-android/pull/5459/files#diff-2afef309affdf8a1c7137b6d8dc5a6f759f3fc35faac96ba473fbb0a89397d6cR110)
2. [testRunWithCoverageAsync_coverageDataMissing_throwsException](https://github.com/oppia/oppia-android/pull/5459/files#diff-2afef309affdf8a1c7137b6d8dc5a6f759f3fc35faac96ba473fbb0a89397d6cR147)

Also, thanks @BenHenning for the suggestion on reproducing the cases, I believe this should suffice, but I’ll refine the approach if necessary.

Reply:
Sounds good, thanks.